src/opentracing/Tracer.ts

```diff
@@ -0,0 +1,143 @@
+import { Tracer, Reference, Tags } from 'opentracing';
+import * as opentracing from 'opentracing';
+import ThundraSpan, { SpanEvent } from './Span';
+import ThundraSpanContext from './SpanContext';
+import ThundraRecorder from './Recorder';
+import ThundraSampler from './Sampler';
+import Utils from '../plugins/Utils';
+
+class ThundraTracer extends Tracer {
+  tags: any;
+  recorder: ThundraRecorder;
+  sampler: ThundraSampler;
```

Comment:
When there is custom sampler (other than `ThundraSampler`), does this cause exception while assigning?

Reply:
No, our code is transpiled to JS and JS has no types. In the config, user should provide an JS object with `isSampled` method. I will give more detailed example in docs.